remote/remote_state_s3.go

```diff
@@ -423,6 +439,38 @@ func EnableVersioningForS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, t
 	return errors.WithStackTrace(err)
 }
 
+// Enable bucket-wide Server-Side Encryption for the AWS S3 bucket specified in the given config
+func EnableSSEForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	terragruntOptions.Logger.Printf("Enabling bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
+	input := s3.PutBucketEncryptionInput{
+		Bucket:                  aws.String(config.Bucket),
+		ServerSideEncryptionConfiguration: &s3.ServerSideEncryptionConfiguration{
+			ServerSideEncryption: aws.String("aws:kms"),
+			SSEKMSKeyId: aws.String("aws/s3"),
+		},
+	}
+
+	_, err := s3Client.PutBucketEncryption(&input)
+	return errors.WithStackTrace(err)
+}
+
+// Enable bucket-wide Access Logging for the AWS S3 bucket specified in the given config
+func EnableAccessLoggingForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	terragruntOptions.Logger.Printf("Enabling bucket-wide Access Logging on AWS S3 bucket %s", config.Bucket)
+	input := s3.PutBucketLoggingInput{
+		Bucket:                  aws.String(config.Bucket),
+		BucketLoggingStatus: &s3.BucketLoggingStatus{
+		LoggingEnabled: &s3.LoggingEnabled{
+			TargetBucket: aws.String(config.Bucket),
```

Comment:
Are you writing the access logs back to the same bucket? If so, may be worth mentioning that in the log statement.

Reply:
Yes, in the same bucket. OK - will do.

Reply:
@brikis98 Done.